### PR TITLE
feat: Support different commit message on tags

### LINKED ISSUE #85: Feature Request: Optional tag commit message
We have set the commit message of the version bump to include [skip-ci] which ensures that our main branch push workflow doesn't execute in a continuous loop when the package.json file with the new version is committed.

However Im finding that our downstream workflow that is triggered when the tag is created does not execute. I think this is because by default the same commit message including [skip-ci] is used when creating the tag.

This action should allow a tag commit message to be optionally specified to allow [skip-ci] to be removed and this trigger any downstream workflows.

### PR DESCRIPTION
**Pull request by vodlogic**
Resolves #85. I've tested that the `git show <tag-name>` command displays the new commit message

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -121,6 +123,9 @@ Toolkit.run(async tools => {
     const remoteRepo = `https://${process.env.GITHUB_ACTOR}:${process.env.GITHUB_TOKEN}@github.com/${process.env.GITHUB_REPOSITORY}.git`
     if (process.env['INPUT_SKIP-TAG'] !== 'true') {
       await tools.runInWorkspace('git', ['tag', newVersion])
+      if(tagCommitMessage) {
+        await tools.runInWorkspace('git', ['tag', newVersion, `${newVersion}^{}`, '-f', '-m', `"${tagCommitMessage.replace(/{{version}}/g, newVersion)}"`])
```
Comment (phips28) [inline index.js:127]: @vodlogic  what git command does that produce in your tests?

File: index.js [hunk 2]
```diff
@@ -121,6 +123,9 @@ Toolkit.run(async tools => {
     const remoteRepo = `https://${process.env.GITHUB_ACTOR}:${process.env.GITHUB_TOKEN}@github.com/${process.env.GITHUB_REPOSITORY}.git`
     if (process.env['INPUT_SKIP-TAG'] !== 'true') {
       await tools.runInWorkspace('git', ['tag', newVersion])
```
Comment (phips28) [inline index.js:125]: now you add 2 tags? do you want an if-else here?

Comment (vodlogic) [reply index.js:125]: No it updates the commit message on the specified tag

See https://stackoverflow.com/questions/7813194/how-do-i-edit-an-existing-tag-message-in-git/23532519

File: index.js (hunk 1 above)
Comment (vodlogic) [reply index.js:127]: git tag v8.2.6 v8.2.6\^{} -f -m "new tag message"

File: index.js (hunk 2 above)
Comment (phips28) [reply index.js:125]: ah ok thats cool, didnt know that exists 🙂

Comment (vodlogic) [reply index.js:125]: I've just had a thought about an edge case where if only the `tag-commit-message` input is specified then any subscribed downstream workflows would be triggered twice as the original commit message does not include [skip-ci] by default.

Perhaps reverting to a simpler if/else would be better to address this problem. What do you think?

Comment (phips28) [reply index.js:125]: ok, but thats only for devs that use it this way.

no existing repos would break right?

File: index.js [hunk 3]
```diff
@@ -121,6 +123,9 @@ Toolkit.run(async tools => {
...
+      if(tagCommitMessage) {
```
Comment (phips28) [inline index.js:126]: code formatting `if (`

**Review by phips28 — CHANGES_REQUESTED** [review]

**Comment by vodlogic** [comment]
So with more testing on github actions itself I can confirm that using another commit message on the tag using both methods of specifying a message without [cskip-ci] still doesn't trigger another workflow. I have concluded that the original commit message is always used by github and that matches similar behaviour with Circle CI. Im happy to close this PR without merging.

**Pull request closed without merging**